frontend/src/services/transactions/token.ts

```diff
@@ -21,21 +21,25 @@ export async function ensureTokenAllowance(
   }
 }
 
-export async function getTokenDecimals(
-  ethereumProvider: Readonly<EthereumProvider>,
-  tokenAddress: string,
-): Promise<BigNumber> {
+/**
+ * @returns number of decimals of token, undefined if not connected
+ */
+export async function getTokenDecimals(tokenAddress: string): Promise<BigNumber | undefined> {
```

Comment:
What would be wrong about getting the provider from the store in the place where this function is called and still pass it as a parameter?
I liked that we were not returning undefined values from the functions.

Your initial comment sounds like you already had a discussion about this. Just want to understand the decisions made here.

Reply:
See [this comment](https://github.com/beamer-bridge/beamer/pull/581#issuecomment-1106279404) and the last commit.